app/Core/Repository/ServiceRepository.cs

```diff
@@ -86,6 +86,76 @@ public Services GetService(Services service)
             return null;
         }
 
+        public bool IsDuplicateServiceCode(string serviceCode)
+        {
+            string query = "SELECT COUNT(*) FROM services WHERE serviceCode = @ServiceCode AND isDeleted = 0";
+
+
+            Dictionary<string, object> parameters = new Dictionary<string, object>
+    {
+        { "@ServiceCode", serviceCode }
+    };
+
+
+            UpgradeFile upgradeFile = new UpgradeFile();
+
+
+            int count = upgradeFile.ExecuteScalar(query, parameters);
+
+            return count > 0;
+        }
+
+
+        public bool UpdateServicePrice(string serviceCode, decimal newPrice)
+        {
+            string query = "UPDATE services SET Price = @NewPrice WHERE ServiceCode = @ServiceCode AND isDeleted = 0";
+
+
+            Dictionary<string, object> parameters = new Dictionary<string, object>
+    {
+        { "@NewPrice", newPrice },
+        { "@ServiceCode", serviceCode }
+    };
+
+
+            UpgradeFile upgradeFile = new UpgradeFile();
+
+
+            int rowsAffected = upgradeFile.ExecuteScalar(query, parameters);
+
+            return rowsAffected > 0;
+        }
+
+        public bool UpdateService(app.core.model.Services service)
```

Comment:
Redundant code, improve the existing SaveService method to accommodate the INSERT/UPDATE action of the Service module. See other reference from different modules.